src/ex1/ItemManager.js

```diff
@@ -0,0 +1,60 @@
+class ItemManager{
+    constructor(element) {
+        this.todoList = [];
+        this.pcInstance = new PokemonClient();
+    }
+
+    async addTask(itemText){
+        let itemTexts = [];
+        //If the input includes string, parse it to integer, and fetch its pokemon id 
+        if(parseInt(itemText)){
+            itemTexts = await this.pcInstance.fetchPokemon([parseInt(itemText)]);
+        }
+        //If the input has more than 1 string, split it into an array, fetch the input and create an array with parsed integers which are the pokemons' ids
+        if(itemText.split(`,`).length > 1){
+            itemTexts = await this.pcInstance.fetchPokemon(itemText.split(`,`).map(id => parseInt(id)));
+        }
+        //If the array's empty, then push the string into the array and print the string that was inserted
+        if(itemTexts.length === 0){
+            itemTexts.push(itemText);
+        }
+        //For each element in the itemTexts array, push its value into the main array (todoList), allocate an id to it and set its done attribute to false  
+        itemTexts.forEach(item => {
+            this.todoList.push({value:item, id: (Math.random() + 1).toString(36).substring(7), done: false});
```

Comment:
same here. try to avoid long lines with different logic in it
```suggestion
           let id = Math.random() + ... ;
           this.todoList.push({value:item, id, done: false});
```